src/Akka.Streams.Kafka/Stages/Consumers/Actors/KafkaConsumerActor.cs

```diff
@@ -556,6 +609,30 @@ private void Poll()
                 using (var cts = new CancellationTokenSource(_settings.PollTimeout))
                 {
                     var (polled, exception) = PollKafka(cts.Token);
+                    var assignedAtEnd = _consumer.Assignment.ToImmutableHashSet();
+                    
+                    /*
+                     * Any partitions that were NOT ASSIGNED at the start of the poll BUT ARE ASSIGNED NOW
+                     * got assigned to us mid-poll then. We need to stash these messages until they are requested.
+                     */
+                    var newlyAssigned = assignedAtEnd.Except(currentAssignment);
```

Comment:
Sanity check on partitions here - now I don't think this code should even be hit if my changes to `PollKafka` work (depends on Confluent.Kafka consumer behavior) but if it does, this is our fail-safe for preventing message loss upon new partition assignment.

We check for newly assigned partitions that weren't there when we started AND within that set of newly assigned partitions, we are only interested in newly assigned partitions that have not been requested.